libherokubuildpack/src/buildpack_output/mod.rs

```diff
@@ -502,6 +569,30 @@ where
     }
 }
 
+impl<W> BuildpackOutput<state::Background<W>>
+where
+    W: Write + Send + Sync + 'static,
+{
+    /// Finalize a timer's output
+    ///
+    /// Once you're finished with your long running task, calling this function
+    /// finalizes the timer's output and transitions back to a [`state::Section`].
+    #[must_use]
+    pub fn finish(self) -> BuildpackOutput<state::Section<W>> {
+        let duration = self.state.started.elapsed();
+        let mut io = match self.state.write.stop() {
+            Ok(io) => io,
+            Err(e) => std::panic::resume_unwind(e),
+        };
+
+        writeln_now(&mut io, style::details(duration_format::human(&duration)));
```

Comment:
This is where the newline is added at the end. Everything up to this point prints inline.